src/actions/publish.js

```diff
@@ -129,35 +131,52 @@ async function storeJustPublishedProfileData(
   const profileMeta = getProfile(state).meta;
   const profileFilterPageData = getProfileFilterPageData(state);
 
-  await storeProfileData({
-    profileToken,
-    jwtToken,
-    publishedDate: new Date(),
-    name: getProfileName(state),
-    originHostname: profileFilterPageData
-      ? profileFilterPageData.hostname
-      : null,
-    preset: null, // This is unused for now.
-    meta: {
-      // We don't put the full meta object, but only what we need, so that we
-      // won't have unexpected compatibility problems in the future, if the meta
-      // object changes. By being explicit we make sure this will be handled.
-      product: profileMeta.product,
-      abi: profileMeta.abi,
-      platform: profileMeta.platform,
-      toolkit: profileMeta.toolkit,
-      misc: profileMeta.misc,
-      oscpu: profileMeta.oscpu,
-      updateChannel: profileMeta.updateChannel,
-      appBuildID: profileMeta.appBuildID,
-    },
-    urlPath: predictedUrl,
-    publishedRange:
-      removeProfileInformation &&
-      removeProfileInformation.shouldFilterToCommittedRange
-        ? adjustRange(removeProfileInformation.shouldFilterToCommittedRange)
-        : adjustRange(getCommittedRange(state)),
-  });
+  try {
+    await storeProfileData({
+      profileToken,
+      jwtToken,
+      publishedDate: new Date(),
+      name: getProfileName(state),
+      originHostname: profileFilterPageData
+        ? profileFilterPageData.hostname
+        : null,
+      preset: null, // This is unused for now.
+      meta: {
+        // We don't put the full meta object, but only what we need, so that we
+        // won't have unexpected compatibility problems in the future, if the meta
+        // object changes. By being explicit we make sure this will be handled.
+        product: profileMeta.product,
+        abi: profileMeta.abi,
+        platform: profileMeta.platform,
+        toolkit: profileMeta.toolkit,
+        misc: profileMeta.misc,
+        oscpu: profileMeta.oscpu,
+        updateChannel: profileMeta.updateChannel,
+        appBuildID: profileMeta.appBuildID,
+      },
+      urlPath: predictedUrl,
+      publishedRange:
+        removeProfileInformation &&
+        removeProfileInformation.shouldFilterToCommittedRange
+          ? adjustRange(removeProfileInformation.shouldFilterToCommittedRange)
+          : adjustRange(getCommittedRange(state)),
+    });
+  } catch (e) {
+    if (e.name === 'InvalidStateError') {
+      // It's very likely we are in private mode, so let's catch and ignore it.
+      // We can remove this check once Firefox stops erroring,
+      // see https://bugzilla.mozilla.org/show_bug.cgi?id=1639542
+      console.error(
+        stripIndent`
+          A DOMException 'InvalidStateError' was thrown when storing the profile data to a local indexedDB.
```

Comment:
We even have a Warning component now :) https://github.com/firefox-devtools/profiler/blob/main/src/components/shared/Warning.js. But probably this would probably need some state etc, so I'd rather not do it now, but this is a good idea to show it in a better way to the user. I'll file a follow-up issue about that.

Reply:
I filed #2670 for this.